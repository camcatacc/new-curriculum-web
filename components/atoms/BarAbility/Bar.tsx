// Modules
import React from "react";

// Elements
import { Typography } from "@material-ui/core";
import { motion } from "framer-motion";

// Definitions
import type { Variants } from "framer-motion";

export interface BarProps {
	percentage: number;
	backgroundColor: string;
	fontSize?: number | string;
	fontColor?: string;
	animated?: boolean;
}

// Auxiliary functions and options
const getMotionOptions = (animated: boolean, percentage: number) => {
	if (!animated) return {};
	const parentVariants: Variants = {
		offscreen: {
			width: 0
		},
		onscreen: {
			width: `${percentage}%`,
			transition: {
				type: "tween",
				duration: 0.5,
				when: "beforeChildren"
			}
		}
	};

	const childVariants: Variants = {
		offscreen: {
			opacity: 0
		},
		onscreen: {
			opacity: 1,
			transition: {
				type: "tween",
				duration: 0.3
			}
		}
	};
	return { parentVariants, childVariants };
};

// Element
const Bar = ({ percentage, backgroundColor, fontSize = "1rem", fontColor = "white", animated = false }: BarProps) => {
	const { parentVariants, childVariants } = getMotionOptions(animated, percentage);
	return (
		<div className="bg-gray-300 h-full">
			<motion.div
				variants={parentVariants}
				style={{
					width: `${percentage}%`,
					backgroundColor,
					height: "100%"
				}}
				className="pr-4 h-full flex flex-col justify-center"
			>
				<motion.div variants={childVariants}>
					<Typography variant="body1" align="right" style={{ fontSize: fontSize, color: fontColor }}>{`${percentage}%`}</Typography>
				</motion.div>
			</motion.div>
		</div>
	);
};

export default Bar;
