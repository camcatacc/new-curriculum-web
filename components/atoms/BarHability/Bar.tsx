import { Typography } from "@material-ui/core";
import React from "react";
import { lightGrey } from "../../../styles/customColors";

export const Bar = (props: {
    percentage: number;
    color: string;
    fontSize?: number | string;
}) => {
    return (
        <div style={{ backgroundColor: lightGrey, height: "100%" }}>
            <div
                style={{
                    width: `${props.percentage}%`,
                    backgroundColor: props.color,
                    height: "100%",
                }}
                className="pr-8 h-full flex flex-col justify-center"
            >
                <Typography
                    variant="body1"
                    color="textSecondary"
                    align="right"
                    style={{ fontSize: props.fontSize }}
                >{`${props.percentage}%`}</Typography>
            </div>
        </div>
    );
};
