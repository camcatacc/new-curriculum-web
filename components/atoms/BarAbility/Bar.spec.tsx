// Modules
import React from "react";
import { render } from "@testing-library/react";

// Elements
import Bar from "components/atoms/BarAbility/Bar";

// Definitions
import type { BarProps } from "components/atoms/BarAbility/Bar";

// Default Props
const defaultProps: BarProps = {
	percentage: 40,
	color: "red"
};

// Tests
describe("Bar", () => {
	it("Renders as expected", () => {
		const { container } = render(<Bar {...defaultProps} />);
		expect(container).toMatchSnapshot();
	});
});
