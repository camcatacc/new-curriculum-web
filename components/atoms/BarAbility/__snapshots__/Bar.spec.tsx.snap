// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bar Renders as expected 1`] = `
<div>
  <div
    class="bg-gray-300 h-full"
  >
    <div
      class="sc-bdvvtL jGTAWU pr-4 h-full flex flex-col justify-center"
    >
      <div>
        <p
          class="MuiTypography-root sc-gsDKAQ etFjmY MuiTypography-body1 MuiTypography-alignRight"
          font-size="1rem"
          fontcolor="white"
        >
          40%
        </p>
      </div>
    </div>
  </div>
</div>
`;
